Snippets/CosmosDB/CosmosDB_2/Transactions.cs

```diff
@@ -0,0 +1,182 @@
+﻿using System.Collections.Generic;
+using Microsoft.Azure.Cosmos;
+using NServiceBus;
+using NServiceBus.Persistence.CosmosDB;
+
+class Transactions
+{
+    void ExtractPartitionKeyFromHeaderSimple(PersistenceExtensions<CosmosPersistence> persistence)
+    {
+        #region ExtractPartitionKeyFromHeaderSimple
+
+        var transactionInformation = persistence.TransactionInformation();
+        transactionInformation.ExtractPartitionKeyFromHeader("PartitionKeyHeader");
+
+        #endregion
+    }
+
+    void ExtractPartitionKeyFromHeadersExtractor(PersistenceExtensions<CosmosPersistence> persistence, MyAppsCustomConfigurationHolder appConfig)
+    {
+        #region ExtractPartitionKeyFromHeadersExtractor
+
+        var transactionInformation = persistence.TransactionInformation();
+        transactionInformation.ExtractContainerInformationFromHeaders<MyAppsCustomConfigurationHolder>((headers, config) => new ContainerInformation(headers["ContainerNameHeader"], new PartitionKeyPath(config.PartitionKeyPath)), appConfig);
```

Comment:
I wonder if we should show the state based overloads. They are very advanced and mostly relevant only for people that don't want to allocate closures every time this delegate is invoked. Maybe we can switch to the non-state based overload here and have a dedicated snippet that talks about the closure allocation problem or just leave these APIs to be discovered by advanced developers?

Reply:
agreed. Switching to a simpler snippet.